app/classes/battleMethods.py

```diff
@@ -0,0 +1,26 @@
+from app.classes.knight import Knight
+
+
+def fight(k_1: Knight, k_2: Knight) -> None:
+    attack(k_1, k_2)
+    if_died(k_1)
+    if_died(k_2)
+
+
+def attack(knight_1: Knight, knight_2: Knight) -> None:
+    knight_1.hp -= knight_2.power - knight_1.protection
+    knight_2.hp -= knight_1.power - knight_2.protection
+
+
+def if_died(knight: Knight) -> None:
+    if knight.hp <= 0:
+        knight.hp = 0
+
+
+def results(k1: Knight, k2: Knight, k3: Knight, k4: Knight) -> dict:
```

Comment:
Function names should include actions, rename `results` to `get_results` and change naming of `if_died`